### DISCUSSION TITLE
tc relay as AMS edge?

### DISCUSSION THREAD

**Discussion by ROBERT-MCDOWELL** [zenomt/rtmfp-cpp#17] [Ideas]
Discussion open about to use tcserver as a relay.
as adobe edge is also a full server communicating with each other why not to do the same with tc server?
have an instance of tcserver on every node and have  a special protocol or else communication that consider the node X as the publish origin and node Y/Z/.. a relay?

**Comment by doanquocnam** [comment]
From the tcserver source there are several ways to make edge server.:
E.g run 2 tcserver, the first is tcserver, the second is  tc-edge-server.
1. User ffmpeg command : ffmpeg -i rtmp://tcserver/stream/key -f flv -c copy  rtmp://tc-edge-server/stream/key

2. Also, we can use libavformat, avutil from ffmpeg source code to write the code to do this  by using rtmp protocol
3. We can make a combination of tcpublish & tcconn source code to write a  module to receive media from tcserver and then publish to tc-edge-server.

All these three methods I've already done and they are working very well.
